pkg/imageset/updateinstaller_test.go

```diff
@@ -148,14 +158,25 @@ func testClusterDeploymentWithErrorCondition() *hivev1.ClusterDeployment {
 	return cis
 }
 
-func validateSuccessfulExecution(t *testing.T, clusterDeployment *hivev1.ClusterDeployment) {
-	if clusterDeployment.Status.InstallerImage == nil ||
-		*clusterDeployment.Status.InstallerImage != testInstallerImage {
-		t.Errorf("did not get expected installer image in status")
+func testClusterDeploymentWithInstallerImageOverride(override string) *hivev1.ClusterDeployment {
+	cd := testClusterDeployment()
+	cd.Spec.Provisioning = &hivev1.Provisioning{
+		InstallerImageOverride: override,
 	}
-	condition := controllerutils.FindClusterDeploymentCondition(clusterDeployment.Status.Conditions, hivev1.InstallerImageResolutionFailedCondition)
-	if condition != nil && condition.Status != corev1.ConditionFalse {
-		t.Errorf("unexpected condition status")
+	return cd
+}
+
+func validateSuccessfulExecution(expectedInstallerImage string) func(*testing.T, *hivev1.ClusterDeployment) {
+	return func(t *testing.T, clusterDeployment *hivev1.ClusterDeployment) {
+		if clusterDeployment.Status.InstallerImage == nil {
+			t.Error("did not get an installer image in status")
+		} else if actual := *clusterDeployment.Status.InstallerImage; actual != expectedInstallerImage {
+			t.Errorf("did not get expected installer image in status: got %s, expected %s", actual, expectedInstallerImage)
+		}
```

Comment:
Can we just do some kind of require/assert.NotNil() for the clusterDeployment.Status.InstallerImage == nil check, and and equivalent assert/require for the clusteDeployment.Status.InstallerImage check?

Reply:
That would be my preference, but I was conforming to how the existing code in this file was doing things. I can make a followon PR to cut the whole test suite over to use testify if ya like.

Reply:
works for me

Reply:
https://github.com/openshift/hive/pull/1622/commits/83d20afc84764395899904583a81baea761c97a9